packages/model-viewer/src/features/scene-graph/material.ts

```diff
@@ -64,36 +65,38 @@ export class Material extends ThreeDOMElement implements MaterialInterface {
       gltfMaterial.emissiveFactor = [0, 0, 0];
     }
 
-    let {
+    const {
       normalTexture: gltfNormalTexture,
       occlusionTexture: gltfOcculsionTexture,
       emissiveTexture: gltfEmissiveTexture
     } = gltfMaterial;
 
-    let normalTexture: ThreeTexture|null = null;
-    let occlusionTexture: ThreeTexture|null = null;
-    let emissiveTexture: ThreeTexture|null = null;
-
     const {normalMap, aoMap, emissiveMap} =
         correlatedMaterials.values().next().value;
 
-    if (gltfNormalTexture != null && normalMap != null) {
-      normalTexture = normalMap;
-    } else {
-      gltfNormalTexture = {index: -1};
```

Comment:
The big simplification here was to allow glTF data to be represented as null in the cases where we would want to conjure things up out of thin air. This drastically simplifies all the bookkeeping we need to do to support adding for write operations such as adding a new texture.